### ISSUE TITLE
[Bug] IContext.cs CustomRagTemperature and CustomRagNucleusSampling

### ISSUE THREAD

**Issue by vkilohani** [microsoft/kernel-memory#701]
### Context / Scenario

First of all, thanks for this awesome library. I was trying custom RAG and ran into some issues.

### Issue

Experimenting with custom RAG. Tried to use the IContext interface with:

```
using Microsoft.KernelMemory.Context;

var kernelMemory = .... ; // kernel memory built successfully

var ragContext = RequestContext();
double rag_temp = 0.2; // expected to be a double here: https://github.com/microsoft/kernel-memory/blob/main/service/Core/Search/SearchClient.cs
ragContext.SetArg("custom_rag_temperature_float", rag_temp); // available in the Rag class here: https://github.com/microsoft/kernel-memory/blob/main/service/Abstractions/Constants.cs

string query = ....; // ask a question
answer = kernelMemory.AskAsync(query,  minRelevance: 0.6, context: ragContext);
```

and the program crashes with the following error:

_Unable to cast object of type 'System.Double' to type 'System.Int32'_

Upon inspecting [IContext.cs](https://github.com/microsoft/kernel-memory/blob/main/service/Abstractions/Context/IContext.cs), it seems that the method GetCustomRagTemperatureOrDefault promises to return a double, but the custom value retrieved from context is expected to be of type int, which explains the error. Indeed, by setting rag_temp equal to 1, the program executes successfully.

Similar issue expected with the function right below it, viz. GetCustomRagNucleusSamplingOrDefault (haven't tried but looks exactly like the same issue - promises double, but context tries to get int).

### Suggestion

Am quite new to dotnet, so not absolutely sure about everything I just said. Would be great to have:

1. Consistent types.
2. String names with correct descriptions, e.g. if temp is expected to be a double, then the Temperature string [here](https://github.com/microsoft/kernel-memory/blob/main/service/Abstractions/Constants.cs) could be renamed as "custom_rag_temperature_double".

### Importance

a fix would make my life easier

### Platform, Language, Versions

Programming Language: C#
dotnet version: 8.0.303
Version of Kernel Memory: 0.65.240620.1
OS: Windows 11

### Relevant log output

```shell
Unhandled exception. System.InvalidCastException: Unable to cast object of type 'System.Double' to type 'System.Int32'.

 at Microsoft.KernelMemory.Context.ContextExtensions.TryGetArg[T](IContext context, String key, T& value)
 at Microsoft.KernelMemory.Context.CustomContextExtensions.GetCustomRagTemperatureOrDefault(IContext context, Double defaultValue)
 at Microsoft.KernelMemory.Search.SearchClient.GenerateAnswer(String question, String facts, 
IContext context, CancellationToken token)
 at Microsoft.KernelMemory.Search.SearchClient.AskAsync(String index, String question, ICollection`1 filters, Double minRelevance, IContext context, CancellationToken cancellationToken)
 at Program.<Main>$(String[] args) in C:\..\rag.cs:line 132
 at Program.<Main>(String[] args)
```

**Comment by marcominerva** [comment]
@vkilohani I have made a PR to fix the type inconsistencies in [IContext.cs](https://github.com/microsoft/kernel-memory/blob/main/service/Abstractions/Context/IContext.cs).

**Comment by dluc** [comment]
fixed in #704

**Comment by vkilohani** [comment]
@marcominerva Sorry for my delayed reply & thank you!
